osrf_testing_tools_cpp/src/memory_tools/impl/static_allocator.hpp

```diff
@@ -105,7 +106,14 @@ class StaticAllocator
   }
 
 private:
-  uint8_t memory_pool_[MemoryPoolSize];
+  static std::size_t
+  align_up(std::size_t n) noexcept
+  {
+    return (n+(alignment-1)) & ~(alignment-1);
```

Comment:
Let's make this an inline function so that we can write tests for that this one. Would be nice to make it more readable too, or at least add comments.

Reply:
This trick is used in Boost and shows up all over the place - I don't know that a unit test is really necessary? https://github.com/boostorg/align/blob/develop/include/boost/align/align_up.hpp#L19

Reply:
1. Tests are not there only to make sure your code works. It is there to ensure that the next person can understand it (tests are a type of documentation), modify it easily and trust it.
2. Reading your code, `std::max_align_t` is implementation defined meaning that it may or may not be signed type and may be an integral type of varying length... unlike the original implementation where `alignment` is always a `std::size_t`. Knowing that we will use this code on various architectures with various STL implementations, why take the risk of having this silently start returning bad results?

Reply:
I see your points, I will add to the unit tests.

To elaborate specifically on point 2 though, I am only using `alignof(std::max_align_t)` which the alignof operator does return a `std::size_t` according to the spec, so the type `std::max_align_t` doesn't matter https://en.cppreference.com/w/cpp/language/alignof